Source/Classes/Categories/PINImage+DecodedImage.m

```diff
@@ -224,6 +277,47 @@ UIImageOrientation pin_UIImageOrientationFromImageSource(CGImageSourceRef imageS
     return orientation;
 }
 
+void pin_degreesFromOrientation(UIImageOrientation orientation, void (^completion)(CGFloat degrees, BOOL horizontalFlip, BOOL verticalFlip)) {
+    switch (orientation) {
+        case UIImageOrientationUp: // default orientation
+            completion(0.0, NO, NO);
+            break;
+        case UIImageOrientationDown: // 180 deg rotation
+            completion(180.0, NO, NO);
+            break;
+        case UIImageOrientationLeft:
+            completion(270.0, NO, NO); // 90 deg CCW
+            break;
+        case UIImageOrientationRight:
+            completion(90.0, NO, NO); // 90 deg CW
+            break;
+        case UIImageOrientationUpMirrored: // as above but image mirrored along other axis. horizontal flip
+            completion(0.0, YES, NO);
+            break;
+        case UIImageOrientationDownMirrored: // horizontal flip
+            completion(180.0, YES, NO);
+            break;
+        case UIImageOrientationLeftMirrored: // vertical flip
+            completion(270.0, NO, YES);
+            break;
+        case UIImageOrientationRightMirrored: // vertical flip
+            completion(90.0, NO, YES);
+            break;
+    }
+}
+
 #endif
 
+BOOL pin_CGImageRefIsOpaque(CGImageRef imageRef) {
```

Comment:
Please `NS_INLINE` and move towards the top of the file.

Reply:
Consider it done 👍